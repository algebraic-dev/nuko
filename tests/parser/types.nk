let a : forall a. Int -> a -> String = 2