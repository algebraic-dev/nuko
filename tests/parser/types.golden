✓ That
 └Program
   ├typeDecls: Empty List
   ├letDecls:
   │ └List
   │   └LetDecl
   │     ├declName: (Name "a" 0:4-0:5)
   │     ├declArgs: Empty List
   │     ├declBody: (Lit NoExt (LInt 0:29-0:30 2))
   │     ├declRet:
   │     │ └Just
   │     │   └TArrow
   │     │     ├TId NoExt (Path "Int")
   │     │     ├TArrow
   │     │     │ ├TPoly NoExt (Name "a" 0:15-0:16)
   │     │     │ ├TId NoExt (Path "String")
   │     │     │ └0:15-0:26
   │     │     └0:8-0:26
   │     └declExt: NoExt
   ├impDecls: Empty List
   └programExt: NoExt
